src/raytracer2018.nim

```diff
@@ -1624,33 +1640,20 @@ proc traceAxion(res: var Axion,
 
   vectorBeforeXRT = vectorXRT
   ## Coordinate transform from cartesian to polar at the XRT entrance
-  var
-    vectorEntranceXRTCircular = vec3(0.0)
-  let
-    radius1 = sqrt((pointEntranceXRT[0].mm) *
-      (pointEntranceXRT[0].mm) + (pointEntranceXRT[1].mm) * (pointEntranceXRT[1].mm))
-    phi_radius = arctan2(-pointEntranceXRT[1], (pointEntranceXRT[
-        0])) #arccos((pointEntranceXRT[1]+d) / radius1)
-    alpha = arctan(radius1 / expSetup.detectorInstall.distanceDetectorXRT)
-    phi_flat = radtoDeg(arccos(pointEntranceXRT[0].mm / radius1))
-  vectorEntranceXRTCircular[0] = radius1.float
-  vectorEntranceXRTCircular[1] = phi_radius #in rad
-  vectorEntranceXRTCircular[2] = alpha #in rad
-
-  ### 3D spider structure 1st draft ###
-
-  var factorSpider = (-85.0 - pointExitCB[2]) / vectorXRT[2]
-  var pointEntrancSpider = pointExitCB + factorSpider * vectorXRT
   let
-    radius1Spider = sqrt((pointEntrancSpider[0].mm) *
-      (pointEntrancSpider[0].mm) + (pointEntrancSpider[1].mm) * (pointEntrancSpider[1].mm))
-    phi_flatSpider = radtoDeg(arccos(pointEntrancSpider[0].mm / radius1Spider))
-  #echo phi_flat, " ", phi_flatSpider, " ", pointEntrancSpider
+    (radius1, _) = radiusAndPhi(pointEntranceXRT)
+    vectorEntranceXRTCircular = vec3(
+      radius1.float,
+      # phi
+      arctan2(-pointEntranceXRT[1], (pointEntranceXRT[0])), #arccos((pointEntranceXRT[1]+d) / radius1)
+      # alpha
+      arctan(radius1 / expSetup.detectorInstall.distanceDetectorXRT) # angle
+    ) #in rad
```

Comment:
This is what I do now.

Reply:
ok sounds good but I don't think we need alpha and phi for anything anymore. I could be wrong though

Reply:
Otherwise of course we could replace the vector just by the numbers that we actually use, if that's the case.

Reply:
I wonder if it's again a relict from Christoph's code